Add thermal DEM documentation (#1532)
chaos-polymtl/lethe: doc/source/theory/multiphase/cfd_dem/dem.rst

    \mathbf{x}_{i}^{n+1} &= \mathbf{x}_{i}^{n} + \mathbf{v}_{i}^{n+\frac{1}{2}}dt \\
    \mathbf{v}_{i}^{n+1} &= \mathbf{v}_{i}^{n+\frac{1}{2}} + \mathbf{a}_{i}^{n+1}\frac{dt}{2}

--------------------------------
Thermal DEM in a Stagnant Gas
--------------------------------

> The heat transfer mechanisms considered here are:

Reviewer: ```suggestion
The heat transfer mechanisms considered are:
```

.. math::

   \frac{d T_i}{dt} = \frac{Q_i}{m_i c_i} 

.. math::

   Q_{i j} = H_{i j} (T_j - T_i)

where :math:`Q_i \approx \sum Q_{ij} + Q_{Si}` is the total rate of heat transferred to particle `i`, :math:`Q_{ij}` is the rate of heat transferred between particles `i` and `j`, :math:`Q_{Si}` is a potential source term applied to particle `i` and :math:`H_{ij}` is the thermal conductance between particles `i` and `j`.

The thermal conductance, which is the inverse of the thermal resistance :math:`R_{ij}`, is calculated as follows by Beaulieu *et al*. [#beaulieu2020]_:

.. math::

   \frac{1}{R_{ij}} = \frac{1}{R_L + \left( \frac{1}{R_s} + \frac{1}{R_g} \right)^{-1}} + \frac{1}{R_c + R_G}

Where:

* :math:`R_L` resistance of the contact surface [#batchelor1977]_
* :math:`R_s` resistance of the microcontacts [#vanlew2016]_
* :math:`R_g` resistance of the interstitial gas microgap [#bahrami2006]_
* :math:`R_c` resistance of the solid layers of the particles [#beaulieu2020]_
* :math:`R_G` resistance of the interstitial gas macrogap [#bahrami2006]_

.. figure:: images/particle_particle_resistances.png
    :width: 700
    :align: center

>     Modeling Heat transfer between two rough particles in contact. Inspired by [#beaulieu2020]_.

Reviewer: ```suggestion
    Modeling Heat transfer between two rough particles in contact. Adapted from [#beaulieu2020]_.
```

~~~~~~~~~~~~~~~~~~~~
Thermal Resistances
~~~~~~~~~~~~~~~~~~~~

The thermal resistances, which model the heat transfer between particles are calculated as follows:

.. math::

   R_L &= \frac{1}{2 k_h r_c } \\
   R_s &= \left(\frac{H'}{P_0}\right)^{0.96} \frac{(1+0.96/2)}{1.25 \, \pi \, r_c^2 \, k_h}\left(\frac{\sigma}{\tau}\right) \\
   R_c &= R_{c,i} +R_{c,j}, \quad R_{c,i} = \frac{L_i}{k_i A_i} \\
   R_g &= \frac{2\sqrt{2}\sigma a_2}{\pi k_g r_c^2 \ln\left(1+\frac{a_2}{a_1+M/(2\sqrt{2}\sigma)}\right)} \\
   R_G &= \frac{2}{\pi k_g \left[S \ln\left(\frac{S-B}{S-A}\right) + B - A\right]} \\

The contact radius :math:`r_c` is calculated as follows:

.. math::

>    r_c = \left( \frac{3F_n r^* }{4E^*}\right)^{1/3}

Reviewer: Agreed

     - :math:`a_2`
     - :math:`\operatorname{erfc}^{-1}(0.03P_0/H') - a_1`
   * - Gas molecular mean free path
     - :math:`\Lambda`
     - value depends on the gas
   * - Gas parameter
     - :math:`M`
     - :math:`\left( \frac{2 - \alpha_{T_i}}{\alpha_{T_i}} + \frac{2 - \alpha_{T_j}}{\alpha_{T_j}} \right)\left( \frac{2 \gamma_g}{1 + \gamma_g} \right)\frac{\Lambda}{Pr}`
   * - Gas parameter
     - :math:`S`
     - :math:`2\left(r_h - \frac{r_c^2}{2r_h}\right) + M`
   * - Gas Prandtl number
     - :math:`Pr`
     - :math:`\frac{\mu_g c_g}{k_g}`
   * - Gas specific heats ratio
     - :math:`\gamma_g`
     - value depends on the gas
   * - Geometrical parameter
     - :math:`A`
     - :math:`2\sqrt{r_h^2 - r_c^2}`
   * - Geometrical parameter
     - :math:`B`
     - :math:`0` (for simple cubic packing)
   * - Harmonic mean thermal conductivity
     - :math:`k_h`
     - :math:`\frac{2k_i k_j}{k_i + k_j}`
   * - Maximum Hertzian contact pressure
     - :math:`P_0`
     - :math:`\frac{2E^* \delta_n}{\pi r_c}`
   * - Thermal accommodation coefficients
     - :math:`\alpha_{T_i}, \, \alpha_{T_j}`
     - values depend on the particles and gas

~~~~~~~~~~~~~~~~~~~~~~~

> Particle-wall Contacts

Reviewer 2: This is thermal particle-wall contacts?

Reviewer 2: if so, make the title more specific.

~~~~~~~~~~~~~~~~~~~~~~~

For particle-wall contacts, conduction is mostly computed the same way, except for minor differences.
For the macrocontact and the solid layers resistances, they are only considered for the particle. In the same way,
the interstitial gas macrogap resistance is halved as there is only a macrogap around the particle and not the wall. 

.. math::

   R_L &= \frac{1}{4 k_p r_c } \\
   R_c &= R_{c,p} = \frac{L_p}{k_p A_p} \\

>    R_G &= \frac{1}{2} \, \frac{2}{\pi k_g \left[S \ln\left(\frac{S-B}{S-A}\right) + B - A\right]} \\

Reviewer 3: Should we still use the subscript "i" instead of "p"? Personally, I think it would be clearer.

   * - Gas thermal conductivity
     - :math:`k_g`
     - value depends on the gas
   * - Gas molecular mean free path
     - :math:`\Lambda`
     - value depends on the gas
   * - Gas Prandtl number
     - :math:`Pr`
     - :math:`\frac{\mu_g c_g}{k_g}`
   * - Gas parameter
     - :math:`M`
     - :math:`\left( \frac{2 - \alpha_{T_i}}{\alpha_{T_i}} + \frac{2 - \alpha_{T_j}}{\alpha_{T_j}} \right)\left( \frac{2 \gamma_g}{1 + \gamma_g} \right)\frac{\Lambda}{Pr}`
   * - Gas parameter
     - :math:`S`
     - :math:`2\left(r_h - \frac{r_c^2}{2r_h}\right) + M`
   * - Geometrical parameter
     - :math:`A`
     - :math:`2\sqrt{r_h^2 - r_c^2}`
   * - Geometrical parameter
     - :math:`B`
     - :math:`0` (for simple cubic packing)

~~~~~~~~~~~~~~~~~~~~~~~~~~
Particle-wall Resistances
~~~~~~~~~~~~~~~~~~~~~~~~~~

For particle-wall contacts, conduction is mostly computed the same way, except for minor differences.
For the macrocontact and the solid layers resistances, they are only considered for the particle. In the same way,
the interstitial gas macrogap resistance is halved as there is only a macrogap around the particle and not the wall. 

.. math::

   R_L &= \frac{1}{4 k_j r_c } \\

>    R_c &= R_{c,j} = \frac{L_j}{k_j A_j} \\

Reviewer 3: Why is it "j" instead of "i", I tought "j" was refering to a neighbour particle?

Author: In the figure the heat goes from the wall to the particle, so it bothered me to put from wall to i, when it is usually from i to j

.. [#beaulieu2020] \C. Beaulieu, “Impact de la ségrégation granulaire sur le transfert de chaleur dans un lit rotatif,” (Order No. 28990310), Ph.D. thesis, Polytechnique Montréal, 2020. Available: `<https://www.proquest.com/dissertations-thèses/impact-de-la-ségrégation-granulaire-sur-le/docview/2626891455/se-2>`_\.

.. [#vanlew2016] \J. T. Van Lew, “On thermal characterization of breeder pebble beds with microscale numerical modeling of thermofluid and pebble-pebble interactions,” (Order No. 10158428), Ph.D. thesis, University of California, Los Angeles, 2016. Available: `<https://www.proquest.com/dissertations-theses/on-thermal-characterization-breeder-pebble-beds/docview/1839265662/se-2>`_\.

.. [#bahrami2006] \M. Bahrami, M. M. Yovanovich, and J. R. Culham, “Effective thermal conductivity of rough spherical packed beds,” Int. J. Heat Mass Transf., vol. 49, no. 19–20, pp. 3691–3701, Sep. 2006, doi: `10.1016/j.ijheatmasstransfer.2006.02.021 <https://doi.org/10.1016/j.ijheatmasstransfer.2006.02.021>`_\.

> .. [#zhou2010] \Z. Y. Zhou, A. B. Yu, and P. Zulli, “A new computational method for studying heat transfer in fluid bed reactors,” Powder Technol., vol. 197, no. 1–2, pp. 102–110, Sep. 2010, doi: `10.1016/j.powtec.2009.09.002 <https://doi.org/10.1016/j.powtec.2009.09.002>`_\.

Reviewer 3: Not specifically related to the pr, but in the reference, violano2018 and israelachvili2011 are not used in the text?
